numpy/_core/src/multiarray/alloc.h

```diff
@@ -98,6 +98,11 @@ _npy_init_workspace(
     TYPE *NAME;                                            \
     _npy_init_workspace((void **)&NAME, NAME##_static, (fixed_size), sizeof(TYPE), (size))
 
+/* Same as above, but additionally nulls the workspace */
```

Comment:
This is not a bad idea, but it is not used right now, and `_npy_cinit_workspace` is not defined either, so it cannot work!

Reply:
oops, sorry, was briefly considering and then decided that it should be a follow-up.